core/templates/dev/head/pages/exploration-player-page/layout-directives/progress-nav.directive.ts

```diff
@@ -131,9 +133,18 @@ angular.module('oppia').directive('progressNav', [
           };
 
           $scope.shouldGenericSubmitButtonBeShown = function() {
-            if ($scope.interactionId === 'ItemSelectionInput' &&
+            // Special cases to show submit button:
+            // 1. ItemSelectionInput interaction with maximum selectable
+            //    choices > 1 in desktop/mobile mode.
+            // 2. ItemSelectionInput interaction with maximum selectable
+            //    choices == 1 in mobile mode.
+            // 3. MultipleChoiceInput interaction in mobile mode.
+            if ((BrowserCheckerService.isMobileDevice() &&
+                ($scope.interactionId === 'ItemSelectionInput' ||
+                $scope.interactionId === 'MultipleChoiceInput')) ||
+                ($scope.interactionId === 'ItemSelectionInput' &&
                 $scope.interactionCustomizationArgs
-                  .maxAllowableSelectionCount.value > 1) {
+                  .maxAllowableSelectionCount.value > 1)) {
```

Comment:
I would rewrite this to something like:
```typescript
if (BrowserCheckerService.isMobileDevice()) {
  return SHOW_SUBMIT_INTERACTIONS.indexOf($scope.interactionId) >= 0
} else {
  return ($scope.interactionId === 'ItemSelectionInput' &&
          $scope.interactionCustomizationArgs
            .maxAllowableSelectionCount.value > 1)
}
```

Reply:
Hm, I think this might be an issue because it would never execute the return statement below (see Line 140)

Reply:
But I think the logic is the same: 
if on mobile, show submit if `ItemSelectionInput` or `MultipleChoiceInput`
if on desktop, show submit if `ItemSelectionInput` and selection count > 1

The logic you have now is quite hard to read and understand even, the comment above helps a bit but we still should try to write the code as readable as possible.

Reply:
Yes, I understand. What I meant was the following return statement will not be executed with the suggested change:
```ts
return (doesInteractionHaveNavSubmitButton() && (
    interactionIsInline ||
    !$scope.canWindowShowTwoCards()
));
```

Reply:
Ah, I see, In that case, I think it would make sense to separate these into a new function and just call the function from here.

Reply:
Done.